documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -0,0 +1,252 @@
+// assembly-client-config.adoc
+
+[id='con-consumer-config-properties-{context}']
+= Kafka consumer configuration tuning
+
+Use a basic consumer configuration with optional properties that are tailored to specific use cases.
+
+When tuning your consumers your primary concern will be ensuring that they cope efficiently with the amount of data ingested.
+As with the producer tuning, be prepared to make incremental changes until the consumers operate as expected.
+
+== Basic consumer configuration
+
+Connection and deserializer properties are required for every consumer.
+Generally, it is good practice to add a client id for tracking.
+
+In a consumer configuration, irrespective of any subsequent configuration:
+
+* The consumer fetches from a given offset and consumes the messages in order, unless the offset is changed to skip or re-read messages.
+* The broker does not know if the consumer processed the responses, even when committing offsets to Kafka because the offsets might be sent to a different broker in the cluster.
+
+[source,shell,subs="+quotes,attributes"]
+----
+# ...
+bootstrap.servers=localhost:9092 <1>
+key.deserializer=org.apache.kafka.common.serialization.StringDeserializer  <2>
+value.deserializer=org.apache.kafka.common.serialization.StringDeserializer  <3>
+client.id=my-client <4>
+group.id=my-group-id <5>
+# ...
+----
+<1> (Required) Tells the consumer to connect to a Kafka cluster using a _host:port_ bootstrap server address for a Kafka broker.
+The consumer uses the address to discover and connect to all brokers in the cluster.
+Use a comma-separated list to specify two or three addresses in case a server is down, but it’s not necessary to provide a list of all the brokers in the cluster.
+If you are using a loadbalancer service to expose the Kafka cluster, you only need the address for the service as the availability is handled by the loadbalancer.
+<2> (Required) Deserializer to transform the bytes fetched from the Kafka broker into message keys.
+<3> (Required) Deserializer to transform the bytes fetched from the Kafka broker into message values.
+<4> (Optional) The logical name for the client, which is used in logs and metrics to identify the source of a request. The id can also be used to throttle consumers based on processing time quotas.
+<5> (Conditional) A group id is _required_ if the consumer belongs to a consumer group.
+
+Consumer groups are used to share a typically large data stream generated by multiple producers from a given topic.
+Consumers are grouped using a `group.id`, allowing messages to be spread across the members.
+
+== Scaling data consumption using consumer groups
+
+Consumer groups share a typically large data stream generated by one or multiple producers from a given topic.
+Consumers with the same `group.id` property are in the same group.
+One of the consumers in the group is elected leader and decides how the partitions that the consumers are interested in should be assigned to the consumers.
```

Comment:
Would "subscribed to" be better than "interested in"?

Reply:
Changed to:
_One of the consumers in the group is elected leader and decides how applicable partitions are assigned to the consumers._